.eslintrc.js

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module.exports = {
+    plugins: ['@tanstack/query', 'unused-imports', 'prettier', 'header'],
+    extends: [
+        'eslint:recommended',
+        'react-app',
+        'plugin:@tanstack/eslint-plugin-query/recommended',
+        'prettier',
+        'plugin:prettier/recommended',
+    ],
+    settings: {
+        react: {
+            version: '18',
+        },
+    },
+    root: true,
+    ignorePatterns: ['node_modules', 'build', 'dist', 'coverage'],
+    rules: {
+        'no-case-declarations': 'off',
+        'no-implicit-coercion': [
+            2,
+            { number: true, string: true, boolean: false },
+        ],
+        '@typescript-eslint/no-redeclare': 'off',
+        '@typescript-eslint/ban-types': [
+            'error',
+            {
+                types: {
+                    Buffer: 'Buffer usage increases bundle size and is not consistently implemented on web.',
+                },
+                extendDefaults: true,
+            },
+        ],
+        'no-restricted-globals': [
+            'error',
+            {
+                name: 'Buffer',
+                message:
+                    'Buffer usage increases bundle size and is not consistently implemented on web.',
+            },
+        ],
+        'header/header': [
+            2,
+            'line',
+            [
+                ' Copyright (c) Mysten Labs, Inc.',
+                ' SPDX-License-Identifier: Apache-2.0',
+            ],
+        ],
+    },
+    overrides: [
+        {
+            files: ['apps/explorer/**/*'],
```

Comment:
explorer and wallet had a lot of custimization in their rules, so they have fairly large override blocks, I didn't want to figure out what each rules does and if it made sense to enable it everywhere in the PR, so leaving those mostly as if (I moved a couple up to apply everywhere, like the license header rule)